### ISSUE TITLE
Layout does not work when edges exist in the model that are along the heirarchy tree.

### ISSUE THREAD

**Issue by finger563** [cytoscape/cytoscape.js-cose-bilkent#48]
I.e. edges from a parent node to one of its children, grand-children, etc. or the reverse.
Example:
![image](https://user-images.githubusercontent.com/213467/28186400-9eb4914e-67e0-11e7-8bbf-a5d446baf11d.png)

When at least one such edge exists in the model, the layout does not do anything (i.e. no nodes are moved).

**Comment by ugurdogrusoz** [comment]
Correct, we consider such edges illegal and do not do anything. You could remove such edges prior to layout and restore them back when layout is finished.

**Comment by finger563** [comment]
Hmm, wouldn't it be easier (since they don't affect layout much) for that to happen internally within the layout routine? :)

**Comment by ugurdogrusoz** [comment]
CoSE is a compound graph layout algorithm and such edges are illegal for compound graphs by definition. Feel free to make the necessary changes and make a PR though.
